File: task-03/Elixir/subtask-03.exs
IO.write("Enter a number: ")
input = IO.gets("")
n = String.to_integer(String.trim(input))

if rem(n, 2) == 0 do
  n = n + 1
end

mid = div(n + 1, 2)

for i <- 1..mid do
  spaces = String.duplicate(" ", mid - i)
  stars = String.duplicate("*", 2 * i - 1)
  IO.write(spaces)
  IO.puts(stars)
end

for i <- (mid - 1)..1 do
  spaces = String.duplicate(" ", mid - i)
  stars = String.duplicate("*", 2 * i - 1)
  IO.write(spaces)
  IO.puts(stars)
end

File: task-03/Rust/subtask-03.rs
use std::io;

fn main() {
    let mut input = String::new();
    println!("Enter a number: ");
    io::stdin().read_line(&mut input).unwrap();

    let n: u32 = input.trim().parse().unwrap();
    let n = if n % 2 == 0 { n + 1 } else { n };
    let mid = (n + 1) / 2;

    for i in 1..=mid {
        let spaces = " ".repeat((mid - i) as usize);
        let stars = "*".repeat((2 * i - 1) as usize);
        println!("{}{}", spaces, stars);
    }

    for i in (1..mid).rev() {
        let spaces = " ".repeat((mid - i) as usize);
        let stars = "*".repeat((2 * i - 1) as usize);
        println!("{}{}", spaces, stars);
    }
}
